sentry-actix/src/lib.rs

```diff
@@ -68,221 +71,267 @@
+    pub fn builder() -> SentryBuilder {
+        Sentry::new().into_builder()
     }
 
     /// Converts the middleware into a builder.
-    pub fn into_builder(self) -> SentryMiddlewareBuilder {
-        SentryMiddlewareBuilder { middleware: self }
+    pub fn into_builder(self) -> SentryBuilder {
+        SentryBuilder { middleware: self }
     }
+}
 
-    fn new_hub(&self) -> Arc<Hub> {
-        Arc::new(Hub::new_from_top(Hub::main()))
+impl Default for Sentry {
+    fn default() -> Self {
+        Sentry::new()
     }
 }
 
-impl Default for SentryMiddleware {
-    fn default() -> Self {
-        SentryMiddleware::new()
+impl<S, B> Transform<S> for Sentry
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S::Future: 'static,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type Transform = SentryMiddleware<S>;
+    type InitError = ();
+    type Future = Ready<Result<Self::Transform, Self::InitError>>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        ok(SentryMiddleware {
+            service,
+            inner: self.0.clone(),
+        })
     }
 }
 
-fn extract_request<S: 'static>(
-    req: &HttpRequest<S>,
+/// The middleware for individual services.
+pub struct SentryMiddleware<S> {
+    service: S,
+    inner: Rc<Inner>,
+}
+
+impl<S, B> Service for SentryMiddleware<S>
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S::Future: 'static,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>>>>;
+
+    fn poll_ready(
+        &mut self,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        self.service.poll_ready(cx)
+    }
+
+    fn call(&mut self, req: ServiceRequest) -> Self::Future {
+        let inner = self.inner.clone();
+        let hub = Arc::new(Hub::new_from_top(Hub::main()));
+        let client = hub.client();
+        let with_pii = client
+            .as_ref()
+            .map_or(false, |x| x.options().send_default_pii);
+        let guard = hub.push_scope();
+
+        let (tx, sentry_req) = sentry_request_from_http(&req, with_pii);
+        hub.configure_scope(|scope| {
+            scope.add_event_processor(Box::new(move |event| {
+                process_event(event, tx.clone(), &sentry_req)
+            }))
+        });
+
+        req.extensions_mut().insert(HubWrapper { hub: hub.clone() });
+
+        let fut = self.service.call(req);
+
+        async move {
+            // Service errors
+            let mut res: Self::Response = match fut.await {
+                Ok(res) => res,
+                Err(e) => {
+                    if inner.capture_server_errors {
+                        hub.capture_event(sentry::event_from_error(&e));
```

Comment:
Is `actix_web::Error` a `std::error::Error`? If so, you can just use `sentry::capture_error()` here?